components/CodeInput/ValidatableCodeInput.tsx

```
@@ -0,0 +1,111 @@
+        const [codeValidationResult] = await Promise.all([
+          codeValidationPromise,
+          successAnimationPromise,
+        ]);
+        resolve(codeValidationResult);
+      } catch (error) {
+        if (error instanceof Error && error.message === 'WrongCode')
+          return resolve(false);
+
+        throw error;
+      }
+    });
```

Biome

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)